symcrypt/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+use core::ffi::c_void;
+use std::mem;
+use std::ptr;
+use symcrypt_sys;
+
+pub const SHA256_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA256_RESULT_SIZE as usize;
+pub const SHA384_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA384_RESULT_SIZE as usize;
+
+pub struct SymCryptInit;
+impl SymCryptInit {
+    pub fn new() {
+        unsafe {
+            //symcrypt_sys::SymCryptInit();
+            // TODO: Find out why SymCryptInit() breaks on linux / windows (BREAKING)
```

Comment:
>// TODO: Find out why SymCryptInit() breaks on linux / windows (BREAKING) [](http://example.com/codeflow?start=12&length=73)

What do you mean by "breaks"? Is it crashing? What's the error?

Reply:
There is a link error and the rest of the code will not compile and or run. We can debug offline regarding this one

Reply:
Is SymCryptInit being exported from the DLL? The last we looked I didn't think it was exported??

Reply:
Oh yeah, that's a good point. SymCryptInit is for statically linked libraries. With the DLL, you should use SymCryptModuleInit instead.